File: app/src/main/java/com/example/sampleapplication/StudentEnrolledCoursesActivity.java
package com.example.sampleapplication;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.sampleapplication.student.StudentCourseAdapter;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

}

public class StudentEnrolledCoursesActivity extends AppCompatActivity {

    ImageView logout;
    RecyclerView student_recylerlview;
    private DatabaseReference mDatabase;
    StudentCourseAdapter studentCourseAdapter;
    ProgressDialog progressDialog;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_student_enrolled_courses);
        logout = findViewById(R.id.id_logout);
        student_recylerlview = findViewById(R.id.student_recylerlview);
        student_recylerlview.setLayoutManager(new LinearLayoutManager(this));
        ArrayList<String> course = new ArrayList<>();
        if (CommonUtils.isConnectedToInternet(StudentEnrolledCoursesActivity.this)) {
            try {
                updatePassword();
                getStudentCourses(course);
            } catch (Exception e) {
                e.printStackTrace();
            }
        } else {
            Toast.makeText(StudentEnrolledCoursesActivity.this, "No Internet Connection....", Toast.LENGTH_SHORT).show();
        }
        studentCourseAdapter = new StudentCourseAdapter(this, course);
        student_recylerlview.setAdapter(studentCourseAdapter);
    
        logout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                FirebaseAuth.getInstance().signOut();
                Intent intent = new Intent(StudentEnrolledCoursesActivity.this, StudentLogin.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
                startActivity(intent);
                finish();
            }
        });
    }

 

   
    
    @Override
    public void onBackPressed() {
    
    }



    

File: app/src/main/java/com/example/sampleapplication/StudentEnrolledCoursesActivity.kt
package com.example.sampleapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import com.google.firebase.auth.FirebaseAuth

class StudentEnrolledCoursesActivity : AppCompatActivity() {

    lateinit var javaSelected:TextView
    lateinit var logout:ConstraintLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_student_enrolled_courses)

        javaSelected=findViewById(R.id.javaSelected);
        logout=findViewById(R.id.logout);

        javaSelected.setOnClickListener {
            val intent = Intent(this, StudentOptionsActivity::class.java)
            startActivity(intent)
        }

        logout.setOnClickListener {
            FirebaseAuth.getInstance().signOut()
            val intent = Intent(this, StudentLogin::class.java)
            startActivity(intent)
        }
    }

    override fun onBackPressed() {
       // super.onBackPressed()
    }
}